gcc/config/sol2.h

```diff
@@ -197,8 +197,15 @@ along with GCC; see the file COPYING3.  If not see
 #undef STARTFILE_ARCH_SPEC
 #define STARTFILE_ARCH_SPEC \
   "%{!shared:%{!symbolic: \
-     %{ansi|std=c*|std=iso9899\\:199409:values-Xc.o%s; :values-Xa.o%s} \
-     %{std=c90|std=gnu90:values-xpg4.o%s; :values-xpg6.o%s}}}"
```

Comment:
I probably missed something, but it seems like this is the line that's wrong.  I think old line 200 should devolve into doing what you want.

Reply:
The two clauses here are independent:

```
%{ansi|std=c*|std=iso9899\\:199409:values-Xc.o%s; :values-Xa.o%s}`
```
Selects between Xc and Xa 

```
%{std=c90|std=gnu90:values-xpg4.o%s; :values-xpg6.o%s}}}
```

Selects between xpg4 and xpg6

With this configuration, you always get two values, from this set: {Xc,Xa} {xpg4,xpg6} and we don't want any xpg* values in the c89/c90 configurations if we are aiming for the same as illumos gcc7.

Reply:
Right.  I was saying I think that the first of those is fine, and could be left alone.  I think?